### ISSUE TITLE
Beta private cluster submodule does not support Dual stack networking

### ISSUE THREAD

**Issue by maximumG** [terraform-google-modules/terraform-google-kubernetes-engine#1656]
### TL;DR

The `beta-private-cluster` submodule lacks support for dual stack networking. It only propose to create a cluster in single stack mode, meaning IPv4 only.

I'd like to introduce cluster with dual stack networking (IPv4/IPv6) as the `google_container_cluster` terraform resource is ready for Ipv4/IPv6. As such, pods will be addressed using IPv4 and IPv6 and Kubernetes Service will follow the same mechanism.

### Terraform Resources

The `ip_allocation_policy` map of the `google_container_cluster` terraform resource already propose the IPv6 stack activation as per the following [documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster)

### Detailed design

I could propose to introduce a new submodule variable `enable_ipv6` to activate the dual stack for both pods IP and Kubernetes Service IP.

**Comment by adamhgriffith-uofu** [comment]
This is being actively worked on in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1672.

**Comment by adamhgriffith-uofu** [comment]
This is being actively worked on in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1672.
